p2p/sender.go

```diff
@@ -223,10 +223,19 @@ func SendReceive(ctx context.Context, tcpNode host.Host, peerID peer.ID,
 		return errors.Wrap(err, "close write", z.Any("protocol", s.Protocol()))
 	}
 
+	zeroResp := proto.Clone(resp)
```

Comment:
what if `resp` is NOT a zero/empty response?

Reply:
true, that is an edge-case not handled by this code

Reply:
but that would be the code that uses this function. We can look at adding some build-time checks that all passed in responses are zero.

Reply:
I added a `isZeroProto` check to the SendReceive function.